src/demo/renderer.js

```diff
@@ -248,17 +249,49 @@ const drawMovingFish = state => {
 };
 
 // Draw a prediction to the canvas.
-const drawPrediction = (predictedClassId, text, x, y, ctx) => {
-  const centeredX = x + constants.fishCanvasWidth / 2;
-  const doesLike = predictedClassId === ClassType.Like;
-
-  ctx.fillStyle = doesLike ? 'green' : 'red';
-  ctx.font = '20px Arial';
-  ctx.textAlign = 'center';
-  if (!doesLike) {
-    text = 'not ' + text;
+const drawPrediction = (state, predictedClassId, ctx) => {
+  let scannerImg;
+  const imgKey =
+    predictedClassId === ClassType.Like ? greenScanner : redScanner;
+
+  if (botImages[imgKey]) {
+    scannerImg = botImages[imgKey];
+  } else {
+    loadImage(imgKey).then(img => {
+      botImages[imgKey] = img;
+      scannerImg = img;
+    });
+    return;
+  }
+
+  const x = state.canvas.width / 2 - scannerImg.width / 2;
+  const y = botY + 50;
+  ctx.drawImage(scannerImg, x, y);
+};
+
+// Draw AI bot to canvas for predict mode.
+const drawPredictBot = state => {
+  if (!botImages.closed) {
+    loadImage(aiBotClosed).then(img => (botImages.closed = img));
+    return;
+  }
+
+  let img = botImages.closed;
+  let botX = state.canvas.width / 2 - img.width / 2;
+  botY = botY || state.canvas.height / 2 - img.height / 2;
+
+  // Move AI bot above fish parade.
+  if (state.isRunning) {
+    botYDestination = botYDestination || botY - 130;
+
+    const distToDestination = Math.abs(botYDestination - botY);
+    if (distToDestination > 1) {
+      const direction = distToDestination === botYDestination - botY ? 1 : -1;
+      botY += direction * botVelocity;
```

Comment:
The remainder of the animation system has a principle that it relies on the passage of time rather than the number of updates to determine where an item should be on screen.  While not crucial here, it would be nice if this movement did the same thing.  This would be done by interpolating between the source and destination locations based on time passed.

Reply:
ah yeah, i had that solution initially and was having trouble getting the movement consistent (definitely a bug somewhere i couldn't find), so i'll leave as-is for now and refactor later